test/shopify-cli/commands/help_test.rb

```diff
@@ -25,10 +41,59 @@ def test_default_behavior_lists_tasks
         end
         output = io.join
 
-        assert_match('Available commands', output)
+        assert_match('Available core commands:', output)
         assert_match(/Usage: .*shopify/, output)
       end
 
+      def test_core_commands
+        io = capture_io do
+          run_cmd('help')
+        end
+        output = io.join
+
+        assert_match('Available core commands:', output)
+        assert_match(/Usage: .*shopify/, output)
+      end
+
+      def test_local_commands_available_within_a_project
+        Project.stubs(:current_project_type).returns('rails')
+        Registry.add(->() { Rails::Commands::Fake }, 'fake_rails')
```

Comment:
@mkevinosullivan I've updated this to make use of the new behaviour you've added, but I realise there's maybe some inconsistency.

To add a new core command, we call `Commands.register`, which then calls `Registry.add`.
To add a new local command, we call `Registry.add` directly, e.g. as in `ProjectType.register_command`.

I wonder if would make sense to split `Commands.register` into `Commands.register_core` and `Commands.register_local` to make the intentional clearer?

Reply:
It's a fair point.  There's a bit of abstraction missing here.  The `register_command` within `ShopifyCli::ProjectType` knows that the Registry that's within `ShopifyCli::Commands` is of type `CLI::Kit::Registry` and accepts the `add` method/message.  It really should be using `ShopifyCli::Commands.register()` method, rather than calling `Registry.add()` directly.

I'll create a new PR for review to:
- add an `is_core` parameter to the register() method, default to `false`
- update the calls to `register` within `ShopifyCli::Commands` to include `true` at the end
- change the `register_command` method in `ProjectType` to use the Commands.register() method instead (thereby abstracting the underlying Registry).

Makes sense?

Reply:
Sounds good, but I think I'd lean towards a separate `register_core` method, following Fowler's advice for avoiding flag arguments:
https://learning.oreilly.com/library/view/refactoring-improving-the/9780134757681/ch11.xhtml#ch11lev1sec3